### ISSUE TITLE
Styled Components `css` prop does not work in Production

### ISSUE THREAD

**Issue by glekner** [vitejs/vite-plugin-react-swc#50]
using Vite 4.1, and vite-plugin-react-swc 3.1.0

`css` prop works in dev mode, but it does not work in production mode.

the DOM css attribute looks like this `css="[object Object]"`
could be related to `transpileTemplateLiterals` ?

**Comment by ArnaudBarre** [comment]
Do you use a the SWC plugin?

**Comment by ArnaudBarre** [comment]
Without a plugin, I have a current test that works:
- Code: https://github.com/vitejs/vite-plugin-react-swc/blob/main/playground/emotion/src/Button.tsx#L15
- Assertion: https://github.com/vitejs/vite-plugin-react-swc/blob/main/playground/emotion/__tests__/emotion.spec.ts#L15

Can you provide a small reproduction?

**Comment by bengry** [comment]
> 

The issue that OP mentioned is with styled-components, not emotion. See https://github.com/bengry/vite-plugin-react-swc/tree/styled-components-repro for a small reproduction on top of this repo.

**Comment by ArnaudBarre** [comment]
My bad, I was induced in error by the linked issue.

Your code effectively throw an error, but I think this is because this pattern is not allowed (not sure I don't use styled components). This compile to invalid code where `primary` is not defined. The same error happens with the babel plugin.

**Comment by bengry** [comment]
You're right. I'm not sure if we use this pattern or not, but regardless it's a bit weird. Anyway, I think the issue is with build time. E.g. if you look at https://github.com/bengry/vite-plugin-react-swc/commit/dde3f0e39dcbe3d5dc191d0836d9255b4be015d8, the issue is simpler.

Reproduction steps:
```bash
pnpm install
cd playground/styled-components
pnpm build
pnpm preview
```
Navigate to the URL on screen and see the component. This is what I see:

![CleanShot 2023-02-13 at 09 18 25@2x](https://user-images.githubusercontent.com/260431/218394787-22b9fe48-fd54-4800-9c5c-2048dd508d74.png)

**Comment by ArnaudBarre** [comment]
I'm using the exact same code and I can't repo. What is your environment?

**Comment by bengry** [comment]
> I'm using the exact same code and I can't repo. What is your environment?

Just to make sure, you're running the _production_ build of my commit, right? I wrote the commands to get a reproduction [here](https://github.com/vitejs/vite-plugin-react-swc/issues/50#issuecomment-1427461658) (I just noticed there was a slight error there - I missed the build step of the plugin itself and added it now).

I also published the output here: https://vite-plugin-reaect-swc-styled-components-repro.vercel.app/, not sure how much it helps though.

I'm using an M1 macOS, with Node v18.13.0. Although I'm not sure how much it matters all that much TBH.

Are you running these steps and seeing a different outcome in a production build of the styled-components plugin?

**Comment by ArnaudBarre** [comment]
Yep I did the exact same steps, I can't repo (node 19, Mac Intel). I'm getting `Button___StyledButton-sc-17626d7-1 imxLCF` for the button class.

I read [the doc](https://styled-components.com/docs/tooling#template-string-transpilation), this is a feature to optimised template literals when targeting browser without support for it, so not a concern in 2023.

**Comment by bengry** [comment]
> Yep I did the exact same steps, I can't repo (node 19, Mac Intel). I'm getting `Button___StyledButton-sc-17626d7-1 imxLCF` for the button class.
> 
> I read [the doc](https://styled-components.com/docs/tooling#template-string-transpilation), this is a feature to optimised template literals when targeting browser without support for it, so not a concern in 2023.

I just tried this on Node 19 as well and got the same results as you.

I think the issue though is that the current version in `main` is not the same as the one that's on npmjs.org. If you [change the version](@vitejs/plugin-react-swc) of `@vitejs/plugin-react-swc` being used to the `latest` on npmjs.org, you should see the issue reproduce.
```console
git clone https://github.com/vitejs/vite-plugin-react-swc
git checkout 896a72433cd5d47b0286fd242ad7baa73f6ec511
cd vite-plugin-react-swc/playground/styled-components
pnpm install
pnpm build
pnpm preview
```

Note that I'm **ignoring the local version of the plugin**, and just using the repo for the playground.

**Comment by ArnaudBarre** [comment]
Oh ok this is an issue that was fixed by https://github.com/vitejs/vite-plugin-react-swc/commit/f52d8707d5d28982f7e58f76280a667f3f44d1da
Available in [v3.2.0](https://github.com/vitejs/vite-plugin-react-swc/releases/tag/v3.2.0)

**Comment by glekner** [comment]
@ArnaudBarre the issue is fixed but we are getting some warnings when using the `css` prop
<img width="627" alt="image" src="https://user-images.githubusercontent.com/24938324/219939879-8a2b814a-3de1-4348-8333-356b82bf4d5c.png">

```
Warning: React.jsx: type is invalid -- expected a string (for built-in components) or a class/function (for composite components) but got: undefined. You likely forgot to export your component from the file it's defined in, or you might have mixed up default and named imports.
```

the warning goes away when not using the `css` prop
